src/test/java/com/googlecode/download/maven/plugin/internal/WGetTest.java

```diff
@@ -53,37 +52,22 @@
  * @author Andrzej Jarmoniuk
  */
 public class WGetTest {
+    @Rule
+    public TemporaryFolder temporaryFolder = new TemporaryFolder();
     private Path cacheDirectory;
-
     private final static String OUTPUT_FILE_NAME = "output-file";
     private Path outputDirectory;
 
     @Before
     public void setUp() throws Exception {
-        cacheDirectory = Files.createTempDirectory("wget-test-cache-");
-        outputDirectory = Files.createTempDirectory("wget-test-");
+        temporaryFolder.create();
```

Comment:
@ajarmoniuk you don't really need to create or delete it. JUnit does it for you automatically

Reply:
> @ajarmoniuk you don't really need to create or delete it. JUnit does it for you automatically

Uhhh, not really:

```
[main] INFO org.eclipse.jetty.server.Server - Started @1478ms

java.lang.IllegalStateException: the temporary folder has not yet been created

	at org.junit.rules.TemporaryFolder.getRoot(TemporaryFolder.java:299)
	at com.googlecode.download.maven.plugin.internal.HttpFileRequesterTest.setUp(HttpFileRequesterTest.java:48)
	at junit.framework.TestCase.runBare(TestCase.java:140)
```

I actually assumed that as well in the beginning, and added the explicit creation and destruction afterwards.

Reply:
Ahh, because we're using a Junit3 test case, which is caused by extending `AbstractMojoTestCase`.

Reply:
Now, removed that inheritance which allowed the rule to work.